clients/rospy/src/rospy/impl/tcpros_base.py

```diff
@@ -613,11 +613,29 @@ def write_header(self):
             return
         fileno = sock.fileno()
         ready = None
-        while not ready:
-            _, ready, _ = select.select([], [fileno], [])
+        poller = None
+        if hasattr(select, 'poll'):
+            poller = select.poll()
+            poller.register(fileno, select.POLLOUT)
+            while not ready:
+                events = poller.poll()
+                for _, flag in events:
+                  if flag & select.POLLOUT:
+                        ready = True
+        else:
+            while not ready:
+                try:
+                    _, ready, _ = select.select([], [fileno], [])
+                except ValueError as e:
+                    logger.error("[%s]: select fileno '%s': %s", self.name, str(fileno), str(e))
+                    raise e
```

Comment:
I believe you can just `raise` and it will automatically re-raise the current exception.